From 0015780e0773b5197e143d4de3337cde1f97b211 Mon Sep 17 00:00:00 2001
From: Serguei Katkov <serguei.i.katkov@intel.com>
Date: Tue, 17 Sep 2013 01:31:11 +0700
Subject: Dalvik: Remove non-strict assert in dvmFillInStackTraceInternal

BZ: 138469

dvmIsSuspended can return false in case Thread is on the way to SUSPEND state.
So theoretically it means that we cannot check whether at this moment the
thread is suspened or not. The comment for the function dvmIsSuspended
states that it is for debugger use-case.
So assert is not strict and should be removed.

Category: device-enablement
Domain: AOSP-Dalvik-Runtime
Origin: internal
Upstream-Candidate: yes

Change-Id: I085ec499bf7e175249827be1f6585bf7d247a1d8
Orig-MCG-Change-Id: I8d103abc625e15de921586c50cca8ce5b5c36e43
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
Reviewed-on: http://android.intel.com:8080/132359
Reviewed-by: Beyler, Jean Christophe <jean.christophe.beyler@intel.com>
Reviewed-by: Lupusoru, Razvan A <razvan.a.lupusoru@intel.com>
Reviewed-by: Chang, Yang <yang.chang@intel.com>
Reviewed-by: Banerji, Udayan <udayan.banerji@intel.com>
Reviewed-by: lab_aqa <lab_aqa@intel.com>
Reviewed-by: Provodin, Vitaly A <vitaly.a.provodin@intel.com>
Reviewed-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Tested-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: Hartley, Timothy D <timothy.d.hartley@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 vm/Exception.cpp |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/vm/Exception.cpp b/vm/Exception.cpp
index af84596..8dbfc13 100644
--- a/vm/Exception.cpp
+++ b/vm/Exception.cpp
@@ -833,8 +833,6 @@ void* dvmFillInStackTraceInternal(Thread* thread, bool wantObject, size_t* pCoun
         *pCount = 0;
     fp = thread->interpSave.curFrame;
 
-    assert(thread == dvmThreadSelf() || dvmIsSuspended(thread));
-
     /*
      * We're looking at a stack frame for code running below a Throwable
      * constructor.  We want to remove the Throwable methods and the
-- 
1.7.4.1

